cpp/src/arrow/ipc/json_simple.cc

```diff
@@ -983,6 +983,16 @@ Status ScalarFromJSON(const std::shared_ptr<DataType>& type,
   return Status::OK();
 }
 
+Status ScalarFromJSON(const std::shared_ptr<DataType>& type,
```

Comment:
These two overloads are not used anywhere, are they?
(they are not even exposed in `json_simple.h`)

Reply:
Removed thanks.